src/js/main.js

```diff
@@ -0,0 +1,225 @@
+  storage.setItem("list", JSON.stringify(todoList));
+  storage.setItem("size", count+"");
+  showItem();
+  inputElement.value="";
+}
+
+function showItem(){
+    let list ="";
+    for(let i of todoList){
+        let com = i.completed == true ? "completed" :"";
+        let check = i.completed == true ? "checked" :"";
+        list +="<li id=li"+i.id+" class="+com+">"
+                    +"<div class=\"view\">"
+                        +"<input class=\"toggle\" id=\""+i.id+"\" type=\"checkbox\""+check+">"
+                        +"<label class=\"label\">"+i.name+"</label>"
+                        +"<button class=\"destroy\" id="+i.id+"></button>"
+                    +"</div>"
+                    +"<input class=\"edit\" value="+i.name+"></li>";
+    }
+    document.getElementById("todo-list").innerHTML =list;
+    
+    const deleteButtons = document.querySelectorAll(".destroy");
+    deleteButtons.forEach(deleteButton => deleteButton.addEventListener("click",deleteItem));
+
+    const viewDivs = document.querySelectorAll(".view")
+    viewDivs.forEach(viewDiv => viewDiv.addEventListener("dblclick",changeMode));
+ 
+    const checkboxs = document.querySelectorAll(".toggle");
+    checkboxs.forEach(checkbox => checkbox.addEventListener("click",changeChecked));
+    
+    const editInputs = document.querySelectorAll(".edit");
+    editInputs.forEach(editInput => editInput.addEventListener("keydown",editKey));
+  
+    showListCount();
+}
+function editKey(event){
+    /*enter key*/
+    if(window.event.key === 'Enter')
+    { 
+        this.setAttribute("value", this.value);
+        const toggleInput = this.parentNode.firstChild.childNodes[1];
+        toggleInput.innerText = this.value;
+        this.parentNode.classList.remove("editing");
+        updateName(this.parentNode.id, toggleInput.innerText)
+        
+        return;
+    }
+    /*ESC key*/
+    if(event.key == 'Escape')
+    {
+        const boforeValue = this.previousSibling.childNodes[1].outerText;
+        this.value = boforeValue;
+        this.parentNode.classList.remove('editing');
+    }
+
+}
+
+function changeChecked(){
+    const checkedInput = this.parentNode.parentNode.classList;
+    const updateId = this.parentNode.parentNode.id;
+    if(checkedInput.contains("completed"))
+    {
+        checkedInput.remove("completed");
+        updateCompleted(updateId, false);
+    }else
+    {
+        checkedInput.add("completed");
+        updateCompleted(updateId, true);
+    }
+}
+
+function updateName(id, name){
+    const realID = id.replace("li","");
+    
+    todoList.forEach( todo =>{
+        if(todo.id === realID)
+        {
+            todo.name = name;  
+        }    
+    });
+    storage.setItem("list", JSON.stringify(todoList));
+    
+}
+
+function updateCompleted(id, completed){
+    const realID = id.replace("li","");
+    
+    todoList.forEach( todo =>{
+        if(todo.id === realID)
+        {
+            todo.completed = completed;  
```

Comment:
변경하고 싶은 completed 값을 외부에서 받는 것도 좋지만
이 경우 같은 값으로도 넘어 올 수도 있고, 이 기능은 true or false만 갖기 때문에 여기서 기존값을 도치 시켜주면 어떨까요?
그럼 파라미터도 줄고, 기존 값과 같은지 체크하는 수고로움이 생기지도 않아 더 간단해질 것 같아요:)

```suggestion
            todo.completed = !todo.completed;  
```

Reply:
맞아요.....! 넵넵 도치 방법 염두해두겠습니다!

감사합니다. 변경했습니다 👍 👍